sympy/polys/rings.py

```diff
@@ -2531,6 +2533,185 @@ def compose(f, x, a=None):
+        Returns
+        =======
+
+            sympy.Number
+                The sparse coefficient of the polynomial at the given symbol and degree.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> _, x, y, z = ring("x, y, z", ZZ)
+        >>> p1 = 2*x**4 + 3*y**4 + 10*z**2 + 3
+        >>> dg = 2
+        >>> sym = 2
+        >>> coeff_wrt(p1, sym, dg)
+        10
+
+        """
+        p1 = self
+        p2 = as_polynomial(p1, {sym})
+        monomial = [0] * len(p1.ring.gens)
+        monomial[sym] = dg
+        monomial = tuple(monomial)
+        return p1.ring(p2[monomial])
+
+
+    def as_polynomial(self, syms):
+        """
+        It converts a polynomial p1 into a sparse representation where the
+        monomials are split into parts with symbols and parts without symbols,
+        and the coefficients are stored accordingly.
+
+        Parameters
+        ==========
+
+            p1 : sympy.Poly
+                The polynomial to compute the sparse coefficients for.
+            syms : set
+                The set of symbols to keep in the sparse coefficients.
+
+        Returns
+        =======
+
+            dict
+                A dictionary of sparse coefficients. The keys are tuples of the symbols
+                that are kept, and the values are dictionaries of the coefficients of
+                the monomials where the symbols are not present.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> _, x, y, z = ring("x, y, z", ZZ)
+        >>> p1 = 2*x**4 + 3*y**4 + 10*z**2 + 3
+        >>> syms = {0}
+        >>> as_polynomial(p1, syms)
+        >>> {(4, 0, 0): {(0, 0, 0): 2}, (0, 0, 0): {(0, 4, 0): 3, (0, 0, 2): 10, (0, 0, 0): 3}}
+
+        """
+        p1 = self
+        num_variables = len(p1.ring.gens)
+        non_symbol_indices = set(range(num_variables)) - syms
+        p2 = defaultdict(dict)
+        variable_range = range(num_variables)
+
+        for monomial, coefficient in p1.items():
+            symbol_indices = set(compress(variable_range, monomial))
+            monomial_with_symbols = [0] * num_variables
+            monomial_without_symbols = [0] * num_variables
+
+            for i in symbol_indices & syms:
+                monomial_with_symbols[i] = monomial[i]
+
+            for i in symbol_indices & non_symbol_indices:
+                monomial_without_symbols[i] = monomial[i]
+
+            p2[tuple(monomial_with_symbols)][tuple(monomial_without_symbols)] = coefficient
+
+        return dict(p2)
+
+
+    def sparse_prem(self, g, x):
+        """
+        Computes the pseudo-remainder of the polynomial `f` with respect to `g` using the sparse representation.
+
+        Parameters
+        ==========
+
+            f (PolyElement): The polynomial to compute the pseudo-remainder for.
+            g (PolyElement): The polynomial to divide `f` by.
+            x (Symbol): The main variable of the polynomials.
```

Comment:
This still does not match the format in the style guide:
https://docs.sympy.org/latest/contributing/docstring.html#parameters-section